tests/cucumber/steps/steps.js

```diff
@@ -1604,6 +1604,28 @@ module.exports = function getSteps(options) {
     );
   });
 
+  Given('expected headers', (headersTable) => {
+    this.expectedHeaders = headersTable.hashes();
```

Comment:
I'm guessing that `headersTable` is an instance of a class that has the `hashes` method. Since JSON serialization + deserialization cannot reconstruct class instances, only basic objects, those methods are lost during browser testing when arguments are passed through the webdriver protocol.

That's likely why the tests are failing in chrome + firefox.

Reply:
Yep, I'm taking a look now...if you have any suggestions on how to handle this let me know. It's pretty unfortunate that the browser tests seem to prevent us from using some builtin features of cucumber.

Reply:
Added the test you suggested.